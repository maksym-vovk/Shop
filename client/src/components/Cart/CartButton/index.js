import React from 'react';

//styles
import './index.scss';

export const CartButton = props => {
    return (
        <button className="cart-button" type="submit">{props.state}</button>
    )
};