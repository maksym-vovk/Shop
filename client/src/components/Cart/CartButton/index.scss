.cart-button {
    width: 270px;
    padding: 15px 0;
    color: white;
    font-size: 18px;
    background-image: linear-gradient(156.8deg, rgba(30, 144, 231, 1) 27.1%, rgba(67, 101, 225, 1) 77.8%);
    border-radius: 5px;
}
