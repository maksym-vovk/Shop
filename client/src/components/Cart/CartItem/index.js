import React, {useState} from 'react';

// styles
import './index.scss'

export const CartItem = (props) => {
  const {quantity, color, image, price, totalItemPrice, name, details, id, changeQuantity, remove} = props;
  let [count, setCount] = useState(quantity);

  const onchangeHandler = (event) => {
    const inputValue = event.target.value;
    const max = 1000;
    const maxLength = max.toString().length - 1;

    const re = /^[0-9\b]+$/;

    if (inputValue < max && (event.target.value === '' || re.test(event.target.value))) {
      setCount(Number(inputValue));
      changeQuantity(Number(inputValue), Number(inputValue * price), id)
    } else {
      parseInt(inputValue.toString().substring(0, maxLength));
    }
  };

  const onDecrementHandler = () => {
    if (count > 1) {
      setCount(--count);
      changeQuantity(count, count * price, id);
    }
  };

  const onIncrementHandler = () => {
    const max = 999;
    if (count < max) {
      setCount(++count);
      changeQuantity(count, count * price, id);
    } else {
      parseInt(count.toString().substring(0, max))
    }
  };

  return (
    <div className="cart-item">
      <img className="cart-item__image"
        src={image}
        alt={name}
      />
      <div className="cart-item__details-block">
        <p className="cart-item__name">{name}</p>
        <p className="cart-item__details">{details}</p>
        <p className="cart-item__details">Color: {color}</p>
      </div>
      <div className="cart-item__price-wrapper">
        <div className="cart-item__input-group">
          <input className="cart-item__input-button"
            type="button"
            onClick={() => {
              onDecrementHandler()
            }}
            defaultValue="-"
          />
          <input className="cart-item__input-value"
            type="number"
            id={id}
            value={count}
            onChange={onchangeHandler}
          />
          <input className="cart-item__input-button"
            type="button"
            onClick={() => {
              onIncrementHandler()
            }}
            defaultValue="+"
          />
        </div>
        <div className="cart-item__price-block">
          <span className="cart-item__price">${totalItemPrice}</span>
          <button className="cart-item__remove-link" onClick={() => remove(id)}>Remove</button>
        </div>
      </div>
    </div>
  )
};
