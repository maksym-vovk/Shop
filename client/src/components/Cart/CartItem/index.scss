input[type=number]::-webkit-inner-spin-button,
input[type=number]::-webkit-outer-spin-button {
  margin: 0;
  -webkit-appearance: none;
}

.cart-item {
  display: flex;
  align-items: center;
  justify-content: space-between;
  margin: 50px 0;
  padding-bottom: 30px;
  border-bottom: 1px solid rgba(210, 215, 211, 1);

  &__image {
    width: 250px;
    height: 250px;
  }

  &__name {
    margin-bottom: 10px;
    font-weight: bold;
    font-size: 24px;
  }

  &__details {
    font-size: 18px;
  }

  &__input-group {
    display: flex;
    height: 50px;
  }

  &__input-button {
    width: 50px;
    padding: 0;
    font-size: 20px;
    background-color: rgba(210, 215, 211, 1);
    cursor: pointer;
  }

  &__input-value {
    width: 50px;
    font-size: 20px;
    text-align: center;
  }

  &__price-block {
    display: flex;
    flex-direction: column;
    margin-left: 30px;
  }

  &__price {
    font-weight: 700;
    font-size: 24px;
  }

  &__remove-link {
    display: flex;
    align-self: center;
    margin-top: 10px;
    color: #337DFF;
    font-size: 18px;
    text-decoration: underline;
    background-color: inherit;
    border: none;
    cursor: pointer;
  }
  &__price-wrapper {
    display: flex;
  }
}

@media (min-width: 1025px) and (max-width: 1280px){
  .cart-item {
    margin: 50px 0;

    &__price-wrapper {
      display: flex;
      flex-direction: column;
      align-items: center;
    }
    &__price-block {
      margin-top: 15px;
      margin-left: 0;
    }
  }
}

/*
  ##Device = Tablets, Ipads (portrait)
  ##Screen = B/w 768px to 1024px
*/
@media (min-width: 768px) and (max-width: 1024px) {
  .cart-item {
    margin: 30px 0;
    &__image {
      width: 230px;
      height: 230px;
    }
    &__name {
      font-size: 21px;
    }
    &__details {
      font-size: 16px;
    }
    &__input-group {
      height: 40px;
    }
    &__input-button {
      width: 40px;
      font-size: 16px;
    }
    &__input-value {
      width: 40px;
      font-size: 16px;
    }
    &__price-wrapper {
      display: flex;
      flex-direction: column;
      align-items: center;
    }
    &__price-block {
      margin-top: 15px;
      margin-left: 0;
    }
    &__price {
      font-size: 22px;
    }
  }
}

@media (min-width: 481px) and (max-width: 767px) {
  .cart-item {
    flex-direction: column;
    margin: 30px 0;
    &__image {
      width: 250px;
      height: 250px;
    }
    &__name {
      font-size: 22px;
    }
    &__details-block {
      margin: 15px 0;
    }

    &__details {
      font-size: 16px;
    }
    &__input-group {
      height: 30px;
    }

    &__input-button {
      width: 30px;
      font-size: 16px;
    }
    &__input-value {
      width: 30px;
      font-size: 16px;
    }
    &__price {
      font-size: 22px;
    }
    &__price-block {
      margin-top: 15px;
      margin-left: 0;
    }
    &__remove-link {
      font-size: 18px;
    }
    &__price-wrapper {
      flex-direction: column;
      align-items: center;
    }
  }
}

@media (min-width: 320px) and (max-width: 480px) {
  .cart-item {
    flex-direction: column;
    margin: 20px 0;

    &__name {
      font-size: 20px;
    }

    &__details-block {
      margin: 15px 0;
    }
    &__details {
      font-size: 16px;
    }
    &__input-group {
      height: 30px;
    }

    &__input-button {
      width: 30px;
    }
    &__input-value {
      width: 30px;
      font-size: 14px;
    }
    &__price-block {
      margin-top: 10px;
      margin-left: 0;
    }
    &__price {
      font-size: 22px;
    }
    &__price-wrapper {
      flex-direction: column;
      align-items: center;
    }
  }
}