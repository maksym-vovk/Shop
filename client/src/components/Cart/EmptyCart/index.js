import React from 'react';
import {Link} from 'react-router-dom';

// styles
import './index.scss';

export const EmptyCart = () => {
  return (
    <div className="container">
      <div className="empty-cart">
        <h1 className="empty-cart__text">Your bag is empty.</h1>
        <Link to="/">
          <button className="empty-cart__button">Continue Shopping</button>
        </Link>
        <img className="empty-cart__image" src="/static/img/cart/apple-cart.jpg" alt="apple-cart"/>
      </div>
    </div>
  )
};