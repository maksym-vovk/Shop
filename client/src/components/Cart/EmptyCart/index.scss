.empty-cart {
  &__text {
    margin: 20px 0;
  }

  &__button {
    align-self: flex-start;
    width: 260px;
    margin-bottom: 20px;
    padding: 15px 0;
    color: #337DFF;
    font-size: 18px;
    background-color: rgba(226,226,226,.7);
    border-radius: 5px;
    cursor: pointer;
  }
  &__image {
    width: 100%;
    margin-top: 20px;
  }
}

@media (min-width: 481px) and (max-width: 767px) {
  .empty-cart {
    height: fit-content;
    &__text {
      font-size: 22px;
    }
    &__button {
      font-size: 16px;
    }
  }
}

@media (min-width: 320px) and (max-width: 480px) {
  .empty-cart {
    height: fit-content;
    &__text {
      font-size: 22px;
    }
    &__button {
      font-size: 16px;
    }
  }
}