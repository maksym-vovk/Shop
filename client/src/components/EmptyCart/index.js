import React from 'react';

import './index.scss'

export const EmptyCart = () => {
  return (
    <div className='wrap-empty-cart'>
      <h1 className='cart-title'> Your cart is empty. </h1>
      <span className='cart-title'>Sign in to see if you have any saved items. Or continue shopping.</span>
      <div className='img-empty-cart'/>
    </div>
  )
};