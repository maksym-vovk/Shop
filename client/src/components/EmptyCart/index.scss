.wrap-empty-cart {
  display: flex;
  flex-direction: column;
  align-items: center;
  padding-top: 50px;
  & .cart-title {
    margin-bottom: 3vh;
  }
  & .img-empty-cart {
    width: 1200px;
    height: 50vh;
    background-image: url("/static/img/cart/apple-cart.jpg");
    background-size: cover;
  }
}