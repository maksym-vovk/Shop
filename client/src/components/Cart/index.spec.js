import React from 'react';
import { CartButton } from './CartButton';
import { CartItem } from './CartItem';
import { EmptyCart } from './EmptyCart';

describe('Cart components', () => {
  it('CartButton component render', () => {
    const wrapper = shallow(<CartButton />);
    expect(wrapper).toMatchSnapshot();
  });
  it('CartItem component render', () => {
    const wrapper = shallow(<CartItem />);
    expect(wrapper).toMatchSnapshot();
  });
  it('EmptyCart component render', () => {
    const wrapper = shallow(<EmptyCart />);
    expect(wrapper).toMatchSnapshot();
  });
});