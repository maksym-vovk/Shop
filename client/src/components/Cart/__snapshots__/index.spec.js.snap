// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart components CartButton component render 1`] = `ShallowWrapper {}`;

exports[`Cart components CartItem component render 1`] = `ShallowWrapper {}`;

exports[`Cart components EmptyCart component render 1`] = `ShallowWrapper {}`;
